### PR TITLE
SectorNordAG: Clicking 'Enter' inside of <input> triggers wrong event.

### PR DESCRIPTION
**Pull request by ZTrotter**
## Proposed change

On a page with both **Save as new draft** and **Submit** buttons, clicking the _Enter_ key inside a `<input>` box triggers the **Save as new draft** button by default. This behaviour is not consistent with the same pages in Znuny6.

In order to fix this behaviour,  the following code could be added to Core.Form.js to handle this situation. 

```
    $('body').on('keydown', 'input', function (Event) {
        if (Event.keyCode == 13) {
            $('form').find(':submit').last().click();
        }
        Event.preventDefault()
    });

```
Additionally, the code that handles the user attempting to submit a form from within a `<textarea>` when `Frontend::RichText` is disabled could be updated to similarly trigger the expected button by changing the logic of how the button is selected. Currently it selects the first element with the type **submit**, which it mistakenly believes to be **save as new draft**, changing this to `last()` instead will trigger the expected button. 

```
    $('body').on('keydown', 'textarea', function (Event) {
        if ((Event.ctrlKey || Event.metaKey) && Event.keyCode == 13) {
            // We need to click() instead of submit(), since click() has
            // a few useful event handlers tied to it, like validation.
            $(this.form).find(':submit').last().click();
        }
        Event.preventDefault()
    });
```

## Type of change

 '1 - 🐞 bug 🐞'

### Replication 
- Navigate to `AgentTicketZoom`
- Click on **Communication** and select **Note**
- Enter some dummy information in both **Subject** and **Text**, click inside of **Subject** and click __Enter__
- You will be presented with the **Add new draft** popup instead of the form submitting

## Checklist

- [x] The code change is tested and works locally.(❗)
- [x] There is no commented out code in this PR.(❕)
- [ ] You improved or added new unit tests.(❕)
- [ ] Local ZnunyCodePolicy passed.(❕)
- [ ] Local UnitTests / Selenium passed.(❕)
- [ ] GitHub workflow CI (UnitTests / Selenium) passed.(❗)

### CODE REVIEW

**Comment by NiklasSchmitt** [comment]
It seems for me that your code is not working correctly. Tested on AgentTicketNote with disabled `Frontend::RichText`.

- Used your code as it is but it happen nothing when I try to type anything into the input or textarea.
- Tried to removed `Event.preventDefault();`: I've entered some text into the textarea and then I've typed a text into the input field and pressed Enter. The "Save as draft"-popup is opened and a second later the form get submitted and the article is saved.

File: var/httpd/htdocs/js/Core.Form.js [hunk 1]
```diff
@@ -293,6 +293,16 @@ Core.Form = (function (TargetNS) {
         $("form input, select, textarea").off('change', FormModified);
     }
 
+    /**
+     * This makes all forms submittable by using Enter inside inputs.
+     */
+    $('body').on('keydown', 'input', function (Event) {
+        if (Event.keyCode == 13) {
+            Event.preventDefault()
```
Comment (NiklasSchmitt) [inline var/httpd/htdocs/js/Core.Form.js]:
```suggestion
            Event.preventDefault();
```

File: var/httpd/htdocs/js/Core.Form.js [hunk 2]
```diff
@@ -302,7 +312,8 @@ Core.Form = (function (TargetNS) {
         if ((Event.ctrlKey || Event.metaKey) && Event.keyCode == 13) {
             // We need to click() instead of submit(), since click() has
             // a few useful event handlers tied to it, like validation.
-            $(this.form).find(':submit').first().click();
+            Event.preventDefault()
```
Comment (NiklasSchmitt) [inline var/httpd/htdocs/js/Core.Form.js]:
```suggestion
            Event.preventDefault();
```

**Review by NiklasSchmitt — APPROVED** [review]
looks good and is working for me, when adding the missing `;` to the function calls ;)

**Comment by jepf** [comment]
@ZTrotter 

Unfortunately we had to revert this PR because it had the side effect of forms without clickable submit buttons not being submittable anymore via Enter key. This is the case e.g. in AdminSystemConfiguration when using the search box and pressing Enter to show all matching results or in AdminCustomerUser searching for a user and pressing Enter.

If you have a solution for this side effect, we're happy to include it.

Thank you.

**Comment by ZTrotter** [comment]
@jepf @NiklasSchmitt 

I have now created a solution to the above side-effect and would like to update the PR with the latest commit, but as the PR has been merged already, I'm not able to do so. Should I create a new PR for this fix or is there an alternative option?

**Comment by jepf** [comment]
@ZTrotter 

👍 
It would be best to create a new PR.

Thank you!

**Comment by NiklasSchmitt** [comment]
Hi @ZTrotter, the best solution is to create a new PR.

**Comment by ZTrotter** [comment]
Perfect, I'll do that.

Thanks!

**Pull request merged**